tests/better_tls.rs

```diff
@@ -34,16 +49,25 @@ pub fn path_building() {
 
         // Set the time to the time of test case generation. This ensures that the test case
         // certificates won't expire.
-        let now = webpki::Time::from_seconds_since_unix_epoch(1_688_651_734);
-
-        let result = ee_cert.verify_for_usage(
-            &[webpki::ECDSA_P256_SHA256], // All of the BetterTLS testcases use P256 keys.
-            roots,
-            intermediates,
-            now,
-            KeyUsage::server_auth(),
-            None,
-        );
+        let now = webpki::Time::from_seconds_since_unix_epoch(1_691_788_832);
+
+        let subject_name =
+            if let Ok(ip_addr) = IpAddrRef::try_from_ascii(testcase.hostname.as_bytes()) {
+                SubjectNameRef::IpAddress(ip_addr)
+            } else {
+                SubjectNameRef::DnsName(DnsNameRef::try_from_ascii_str(&testcase.hostname).unwrap())
+            };
```

Comment:
I think `SubjectNameRef::try_from_ascii` already does this `parse as an ip address if it looks like that, else parse as a dns name` dance for you.

Reply:
nice, thank you. That saves some imports and makes everything tidier. Applied!